### ISSUE TITLE
shellcheck do not recognize `declare -Ar`

### ISSUE THREAD

**Issue by haolian9** [koalaman/shellcheck#1625]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bib/env bash

declare -Ar dict=(
    [a]=b
)
```

#### Here's what shellcheck currently says:
```
Line 3	SC2034: dict appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:

no warns, as there was already `declare -r`

**Comment by koalaman** [comment]
The variable *is* unused. If you use it, then there is no warning:

```
#!/usr/bib/env bash

declare -Ar dict=(
    [a]=b
)
echo "${dict[a]}"
```

**Comment by haolian9** [comment]
then why shellcheck says nothing about `declare -r dict='whatever'`? I not use it either.

I use this in a `config.sh`, it should be not used, but be sourced, that's why i delcare it `readonly`

**Comment by koalaman** [comment]
You are right, ShellCheck said nothing about `readonly dict='whatever'`. That was a bug. It's now fixed in #1573

It did warn about `declare -r dict='whatever'`, just not `readonly dict='whatever'`
